### PR TITLE
Validate correctness of the tests

### PR DESCRIPTION
**Pull request by nguyentruongtho**
To avoid something like this in the future
https://github.com/facebook/buck/pull/896#discussion_r82299410.

Please feel free to contribute better idea for the validation.

### CODE REVIEW

File: scripts/travisci_run.sh [hunk 1]
```diff
@@ -17,6 +17,9 @@ export BUCK_NUM_THREADS=3
 # Make sure that everything builds in case a library is not covered by a test.
 ./bin/buck build --num-threads=$BUCK_NUM_THREADS src/... test/...
 
+# Validate existence of all test rules.
+./bin/buck test --dry-run --all
+
 # Until we get https://github.com/facebook/buck/pull/479, we cannot run the
```
Comment (Coneko) [inline scripts/travisci_run.sh]: This comment seems stale, the PR was merged.

File: scripts/travisci_run.sh [hunk 2]
```diff
@@ -17,6 +17,9 @@ export BUCK_NUM_THREADS=3
 # Make sure that everything builds in case a library is not covered by a test.
 ./bin/buck build --num-threads=$BUCK_NUM_THREADS src/... test/...
 
+# Validate existence of all test rules.
+./bin/buck test --dry-run --all
```
Comment (Coneko) [inline scripts/travisci_run.sh]: `--dry-run` doesn't work.

File: scripts/travisci_run.sh [hunk 3]
```diff
@@ -17,6 +17,9 @@ export BUCK_NUM_THREADS=3
...
+
 # Until we get https://github.com/facebook/buck/pull/479, we cannot run the
 # Go tests in Travis
 ./bin/buck test --num-threads=$BUCK_NUM_THREADS
```
Comment (Coneko) [inline scripts/travisci_run.sh]: I think you should be able to change this to `./bin/buck test --num-threads=$BUCK_NUM_THREADS //...` to have it resolve all the `tests` attributes.

File: scripts/travisci_run.sh (hunk 2 above)
Comment (nguyentruongtho) [inline scripts/travisci_run.sh]: shouldn't it be removed from the docs & code?

**Pull request closed without merging**